server/server.ts

```diff
@@ -760,9 +760,8 @@ function initializeClient(
 		});
 
 		socket.on("search", async (query) => {
-			await client.search(query).then((results) => {
-				socket.emit("search:results", results);
-			});
+			const results = await client.search(query);
```

Comment:
This function seems to be confused, it's already async but then uses promises as well... simplify